Sources/COWMacros/SyntaxExtensions.swift

```diff
@@ -621,19 +765,49 @@ extension SwitchCaseSyntax {
   
 }
 
-private let autoSynthesizingProtocolTypes: Set<String> = [
+let equatableProtocolNames: Set<String> = [
   "Equatable",
-  "Swift.Equatable",
+  "Swift.Equatable"
+]
+
+let hashableProtocolNames: Set<String> = [
   "Hashable",
-  "Swift.Hashable",
+  "Swift.Hashable"
+]
+
+let codableProtocolNames: Set<String> = [
   "Codable",
-  "Swift.Codable",
+  "Swift.Codable"
+]
+
+let encodableProtocolNames: Set<String> = [
   "Encodable",
-  "Swift.Encodable",
+  "Swift.Encodable"
+]
+
+let decodableProtocolNames: Set<String> = [
   "Decodable",
-  "Swift.Decodable",
+  "Swift.Decodable"
 ]
 
+private let autoSynthesizingProtocolTypes: Set<String> =
+  equatableProtocolNames
+    .union(hashableProtocolNames)
+    .union(codableProtocolNames)
+    .union(encodableProtocolNames)
+    .union(decodableProtocolNames)
+
+extension InheritedTypeListSyntax {
+  func containsType(named typeName: String) -> Bool {
```

Comment:
Explicit `internal` access control is recommended. This convention is learned from the Swift Standard Library.